solutions/cf-1536C.mdx

```diff
@@ -0,0 +1,145 @@
+---
+id: cf-1427G
+source: CF
+title: Diluc and Kaeya
+author: Kevin Sheng
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/91520)
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <vector>
+#include <unordered_map>
+#include <algorithm>
+
+using std::cout;
+using std::endl;
+using std::vector;
+
+// https://codeforces.com/problemset/problem/1536/C (input ommited due to length)
+int main() {
+	int test_num;
+	std::cin >> test_num;
+	for (int t = 0; t < test_num; t++) {
+		int len;
+		std::cin >> len;
+		vector<char> str(len);
+		for (char& c : str) {
+			std::cin >> c;
+		}
+		
+		int d_num = 0;
+		int k_num = 0;
+		vector<int> max_pref_chunks;
+		std::unordered_map<int, std::unordered_map<int, int>> prev_ratios;
+		for (char c : str) {
+			if (c == 'D') {
+				d_num++;
+			} else if (c == 'K') {
+				k_num++;
+			}
+			int gcd = std::__gcd(d_num, k_num);
```

Comment:
you can use gcd from the numeric header in c++17

Reply:
don't wanna put an extra #include